src/package-linker.js

```diff
@@ -368,16 +368,20 @@ export default class PackageLinker {
 
     for (const name in peerDeps) {
       const range = peerDeps[name];
-      const patterns = this.resolver.patternsByPackage[name] || [];
-      const foundPattern = patterns.find(pattern => {
-        const resolvedPattern = this.resolver.getResolvedPattern(pattern);
-        return resolvedPattern ? this._satisfiesPeerDependency(range, resolvedPattern.version) : false;
+      const pkgs = this.resolver.getAllInfoForPackageName(name);
+      let foundPattern;
+      pkgs.find(pkg => {
```

Comment:
You can probably use `invariant()` if nothing helps with that warning. That said, with the way I suggested, it shouldn't be a problem.